scripts/digest_traces.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+""" This script processes a traces file: it extracts the requests that seem
+    to be consuming memory.
+
+    ALS.IN and ALS.OUT need to be active for this to work.
+
+    Usage:
+
+        digest_traces.py <traces_file>
+"""
+
+import json
+import os
+import re
+import sys
+
+TRACES_LINE = \
+  ' *\\[ALS\\.(IN|OUT)\\] ({.*}) \\[Watermark:. (\\d+)/ (\\d+)\\].*'
+# Regexp to parse one line of traces
+
+
+class Digester(object):
+    """ The class that takes care of the file parsing and counting """
+
+    def __init__(self, filename):
+        """Initialize. filename is the path to the traces file"""
+
+        self.in_re = re.compile(TRACES_LINE)
+
+        self.last_method = "(elaboration)"
+        self.last_memory_usage = 0
+        # the last method and last memory usage received
+
+        self.methods_to_memory = {}
+        # cumulative memory increase indexed by method id
+
+        self.methods_exception_count = {"(elaboration)": 0}
```

Comment:
You should replace the hard-coded "(elaboration)" by self.last_method

Reply:
done